### PR TITLE
Allow SSO authentication to provide a user secret

### PR DESCRIPTION
**Pull request by summersab**
Implementing PR #24837 from immerda. It's a feature I'd like to have, and he hasn't been active on GitHub since December. So, I'd like to submit the PR.

### CODE REVIEW

**Comment by summersab** [comment]
Failing the DCO check. I edited the files via my browser because I have rather slow internet, and I didn't want to clone the whole repo via CLI. I'll go ahead and try to fix this.

File: lib/private/legacy/OC_User.php [hunk 1]
```diff
@@ -184,7 +186,7 @@ public static function loginWithApache(\OCP\Authentication\IApacheBackend $backe
 					'post_login',
 					[
 						'uid' => $uid,
-						'password' => '',
+						'password' => '$pw',
```
Comment (MichaIng) [inline lib/private/legacy/OC_User.php:189]: You do not want "$pw" to be assigned literally here, hence the single quotes need to be removed.

Comment (summersab) [reply lib/private/legacy/OC_User.php]: Good catch. Sorry about that.

**Comment by MichaIng** [comment]
Would be great if you could squash the commits to clean the history. But let's see what CI says, probably more to add.

**Comment by summersab** [comment]
> Would be great if you could squash the commits to clean the history. But let's see what CI says, probably more to add.

Forgive my ignorance, but I have no idea how to do that. I did some searching, tried a few tutorials, and wasn't successful. It seems like everyone uses git a bit differently . . .

If you could tell me what commands to run, I'll gladly do it. I'm a technical user, but I'm not the most git-savvy.

**Comment by MichaIng** [comment]
This should do it, basically removing the last three commits from the log and creating a new commit with the apparent changes:
```sh
git reset --soft HEAD~3
git commit
git push -f origin HEAD
```

**Review by MichaIng — APPROVED** [review]

**Review by LukasReschke — CHANGES_REQUESTED** [review]
Something seems to have gone wrong with the rebase.

**Comment by MichaIng** [comment]
A rebase onto the latest upstream branch was missing.

Dammit, I resolved it locally and wanted to push them `summersab:master` master, but pushed master itself instead of the local head, so that no changes were there anymore, leading to the PR being closed automatically. And now that it is closed, I have no push permissions anymore to push the local commit 😅 ... Sorry for that. I'll push it to a new branch and open a PR with that.

**Comment by MichaIng** [comment]
#27929

**Comment by MichaIng** [comment]
@summersab 
Btw, for small changes on a single file I usually use the GitHub editor as well. But if other changes need to be done or fixes, to avoid multiple commits, I do it like that on a local clone:
```sh
git add . # add newly done local changes
git commit --amend --no-edit # amend to existing commit (skip --no-edit if you want to change commit text)
git remote add upstream https://github.com/nextcloud/server.git # if not yet done
git fetch upstream
git rebase upstream/master # rebase the commit onto latest upstream master branch, in case it got changes in the meantime
git push -f origin HEAD
```
And when squashing commits like posted above, always good to assure that the intended commits are included:
```sh
git log
```

**Pull request closed without merging**